cli/ingress-controller/main.go

```diff
@@ -413,10 +414,18 @@ func main() {
 	cacheStores.Plugin = kongPluginInformer.GetStore()
 	informers = append(informers, kongPluginInformer)
 
-	kongClusterPluginInformer := kongInformerFactory.Configuration().V1().KongClusterPlugins().Informer()
-	kongClusterPluginInformer.AddEventHandler(reh)
-	cacheStores.ClusterPlugin = kongClusterPluginInformer.GetStore()
-	informers = append(informers, kongClusterPluginInformer)
+	hasKongClusterPlugin, _ := utils.ServerHasGVK(kubeClient.Discovery(),
```

Comment:
In this case, `err != nil` means that the `ServerHasGVK` call was inconclusive. This can happen due to an intermittent network issue, for example.

The consequence of this call failing might be that KongClusterPlugins get (unexpectedly) omitted by KIC for the lifetime of the KIC process. I'm concerned about this.

I recommend not ignoring the second return value here, and considering the `err != nil` condition an unrecoverable (fatal) error.